src/renderers/native/__tests__/ReactNativeEvents-test.js

```diff
@@ -15,29 +15,93 @@ var PropTypes;
 var RCTEventEmitter;
 var React;
 var ReactNative;
+var ReactNativeBridgeEventPlugin;
 var ResponderEventPlugin;
 var UIManager;
 var createReactNativeComponentClass;
 
+// Parallels requireNativeComponent() in that it lazily constructs a view config,
+// And registers view manager event types with ReactNativeBridgeEventPlugin.
+const fakeRequireNativeComponent = (uiViewClassName, validAttributes) => {
+  const getViewConfig = () => {
+    const viewConfig = {
+      uiViewClassName,
+      validAttributes,
+      bubblingEventTypes: {
+        topTouchCancel: {
+          phasedRegistrationNames: {
+            bubbled: 'onTouchCancel',
+            captured: 'onTouchCancelCapture',
+          },
+        },
+        topTouchEnd: {
+          phasedRegistrationNames: {
+            bubbled: 'onTouchEnd',
+            captured: 'onTouchEndCapture',
+          },
+        },
+        topTouchMove: {
+          phasedRegistrationNames: {
+            bubbled: 'onTouchMove',
+            captured: 'onTouchMoveCapture',
+          },
+        },
+        topTouchStart: {
+          phasedRegistrationNames: {
+            bubbled: 'onTouchStart',
+            captured: 'onTouchStartCapture',
+          },
+        },
+      },
+      directEventTypes: {},
+    };
+
+    ReactNativeBridgeEventPlugin.processEventTypes(viewConfig);
+
+    return viewConfig;
+  };
+
+  return createReactNativeComponentClass(uiViewClassName, getViewConfig);
+};
+
 beforeEach(() => {
   jest.resetModules();
 
   PropTypes = require('prop-types');
   RCTEventEmitter = require('RCTEventEmitter');
   React = require('react');
   ReactNative = require('react-native');
+  ReactNativeBridgeEventPlugin = require('ReactNativeBridgeEventPlugin');
   ResponderEventPlugin = require('ResponderEventPlugin');
   UIManager = require('UIManager');
   createReactNativeComponentClass = require('createReactNativeComponentClass');
 });
 
+it('fails if unknown/unsupported event types are dispatch', () => {
+  expect(() => {
```

Comment:
Is it the .receiveTouches that you expect to throw? If so, can we move the other calls out? An expect() failure also throws an exception – so for example, if RCTEventEmitter.register was never called, or the snapshot didn't match, this test would pass. Even better if you can include the full exception message to ensure it's the error you think it is and not a different bug in receiveTouches.

Reply:
Ah! Yes, good suggestion.